src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -170,7 +170,7 @@ module.exports = function<T, P, I, TI, PI, C, CX>(config : HostConfig<T, P, I, T
 
     deferredUpdates,
 
-    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | I | TI | null) {
+    getPublicRootInstance(container : OpaqueNode) : (ReactComponent<any, any, any> | PI | null) {
```

Comment:
Ah yeah. That’d do it. Test case and possible solution implemented in 9a9a4db394ffe9ce8928ad3381656d853735b1c0.